tests/run-internalbench.py

```diff
@@ -58,27 +71,70 @@ def run_tests(pyb, test_dict):
+against something else, use the -t option.  See below for details.
+""",
+        epilog="""\
+The -t option accepts the following for the test instance:
+- unix - use the unix port of MicroPython, specified by the MICROPY_MICROPYTHON
+  environment variable (which defaults to the standard variant of either the unix
+  or windows ports, depending on the host platform)
+- webassembly - use the webassembly port of MicroPython, specified by the
+  MICROPY_MICROPYTHON_MJS environment variable (which defaults to the standard
+  variant of the webassembly port)
+- port:<device> - connect to and use the given serial port device
+- a<n> - connect to and use /dev/ttyACM<n>
+- u<n> - connect to and use /dev/ttyUSB<n>
+- c<n> - connect to and use COM<n>
+- exec:<command> - execute a command and attach to its stdin/stdout
+- execpty:<command> - execute a command and attach to the printed /dev/pts/<n> device
+- <a>.<b>.<c>.<d> - connect to the given IPv4 address
+- anything else specifies a serial port
+
+Options -i and -e can be multiple and processed in the order given. Regex
+"search" (vs "match") operation is used. An action (include/exclude) of
+the last matching regex is used:
+  run-tests.py -i async - exclude all, then include tests containing "async" anywhere
+  run-tests.py -e '/big.+int' - include all, then exclude by regex
+  run-tests.py -e async -i async_foo - include all, exclude async, yet still include async_foo
+""",
+    )
+    cmd_parser.add_argument(
+        "-t", "--test-instance", default="unix", help="the MicroPython instance to test"
+    )
+    cmd_parser.add_argument(
+        "-b", "--baudrate", default=115200, help="the baud rate of the serial device"
+    )
+    cmd_parser.add_argument("-u", "--user", default="micro", help="the telnet login username")
+    cmd_parser.add_argument("-p", "--password", default="python", help="the telnet login password")
+    cmd_parser.add_argument(
+        "-d", "--test-dirs", nargs="*", help="input test directories (if no files given)"
+    )
+    cmd_parser.add_argument(
+        "--denominator", type=int, default=1, help="divide input iterations by this value"
+    )
     cmd_parser.add_argument("files", nargs="*", help="input test files")
     args = cmd_parser.parse_args()
 
     # Note pyboard support is copied over from run-tests.py, not tests, and likely needs revamping
-    if args.pyboard:
-        import pyboard
-
-        pyb = pyboard.Pyboard("/dev/ttyACM0")
-        pyb.enter_raw_repl()
-    else:
-        pyb = None
+    pyb = run_tests_py.get_test_instance(
+        args.test_instance, args.baudrate, args.user, args.password
+    )
 
     if len(args.files) == 0:
-        if pyb is None:
+        if args.test_dirs:
+            test_dirs = tuple(args.test_dirs)
+        elif pyb is None:
             # run PC tests
             test_dirs = ("internal_bench",)
         else:
             # run pyboard tests
             test_dirs = ("basics", "float", "pyb")
```

Comment:
This bit can be removed, it never worked.

The default `test_dirs` should always just be `("internal_bench",)` regardless of whether `pyb` is `None` or not.

Then, it should be that doing `./run-internalbench.py -t a0` works and runs all the internal bench tests.